docs/arch/adr-03.md

```diff
@@ -0,0 +1,115 @@
+
+ADR 3: Position sampling and patches of the sky
+===============================================
+
+Author: Nicolas Tessore  
+Date: 5 February 2021  
+Status: draft
+
+
+Context
+-------
+
+This ADR addresses two related open problems for SkyPy simulations:
+
+- How can we describe the regions in which to sample the positions of e.g.
+  galaxies or supernovae?
+- How can we break up the sampling over large fractions of the sky into smaller
+  chunks?
+
+The second point is particularly relevant when the simulation becomes so large
+that it no longer fits into memory, as well as for parallelisation.
+
+
+Extension to pipelines
+----------------------
+
+This ADR proposes to introduce a new top-level keyword `regions` [alt: `sky`,
+`geometry`, `patches`] into pipelines that describes the geometry of the
+simulation. For example, to include a "rectangular" and a "circular" region:
+
+```yaml
+regions:
+- !rectangular [ ... ]
+- !circular [ ... ]
```

Comment:
Can you clarify what would go inside the `[ ... ]`, I am unsure what this would look like.  Would it be a bounding box for the rectangle, what coordinate system would it use, etc...

Reply:
Although that might be a topic for a different ADR that outlines how the regions are made.

Reply:
Yes, these are only placeholders, there would likely be a couple of functions taking different specifications.